akka-remote/src/main/scala/akka/remote/artery/ArterySettings.scala

```diff
@@ -292,7 +287,15 @@ private[akka] object ArterySettings {
   def getHostname(key: String, config: Config): String = config.getString(key) match {
     case "<getHostAddress>" => InetAddress.getLocalHost.getHostAddress
     case "<getHostName>"    => InetAddress.getLocalHost.getHostName
-    case other              => other
+    case other =>
+      InetAddress.getByName(other) match {
+        case address: Inet4Address =>
+          address.getHostAddress
+        case address: Inet6Address =>
+          "[" + address.getHostAddress + "]"
+        case _ =>
+          other
+      }
```

Comment:
`getByName` is a blocking DNS lookup if the given name is a hostname, so we can't do it like this. sadly the `IPAddressUtil` used internally by `InetAddress` is JDK internal so we can't use that either. We will likely have to have our own logic to detect if a string is ipv6 without brackets _if_ we want auto-bracketing like this.